### YamlMime:ManagedReference
items:
- uid: System.Xaml.XamlInternalException
  id: XamlInternalException
  children:
  - System.Xaml.XamlInternalException.#ctor
  - System.Xaml.XamlInternalException.#ctor(System.String)
  - System.Xaml.XamlInternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Xaml.XamlInternalException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: XamlInternalException
  nameWithType: XamlInternalException
  fullName: System.Xaml.XamlInternalException
  type: Class
  summary: "Undantag som genereras internt inkonsekvenser som kan uppstå under XAML läsning och skrivning i XAML."
  remarks: "Detta undantag vanligtvis bara när någon del av XAML läsaren eller XAML-skrivaren system har utökats av koden och inkonsekvenser eller implementering problem har uppstått som skickades till .NET Framework XAML Services system koden. Undantag kan kommer från tredje part, till exempel parsa XAML och uppstår fel i XAML, utlösa inte XamlInternalException och utlösa i stället andra undantag, till exempel <xref:System.Xaml.XamlParseException>eller <xref:System.Xaml.XamlObjectWriterException>.</xref:System.Xaml.XamlObjectWriterException> </xref:System.Xaml.XamlParseException>       Om du skriver kod som utökar .NET Framework XAML-tjänster, särskilt bör om du skriver koden för en XAML-läsaren eller XAML-skrivaren implementering du behålla denna skillnad mellan XamlInternalException och andra XAML-relaterade undantag. Koden endast utlösa XamlInternalException i fall där ytterligare tillägg av din klass bryter mot vissa principen eller implementering designmönstret som definierar callable koden. Koden utlösa andra XAML-specifika undantag undantag i de fall där din XAML läsaren eller XAML-skrivaren implementering används utan tillägg och används för att läsa in från tredje part XAML, från tredje part XAML-schemat kontext och så vidare."
  syntax:
    content: 'public class XamlInternalException : System.Xaml.XamlException'
  inheritance:
  - System.Exception
  - System.Xaml.XamlException
  implements: []
  inheritedMembers:
  - System.Xaml.XamlException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Xaml.XamlException.LineNumber
  - System.Xaml.XamlException.LinePosition
  - System.Xaml.XamlException.Message
  platform:
  - net462
- uid: System.Xaml.XamlInternalException.#ctor
  id: '#ctor'
  parent: System.Xaml.XamlInternalException
  langs:
  - csharp
  name: XamlInternalException()
  nameWithType: XamlInternalException.XamlInternalException()
  fullName: System.Xaml.XamlInternalException.XamlInternalException()
  type: Constructor
  assemblies:
  - System.Xaml
  namespace: System.Xaml
  summary: "Initierar en ny instans av den <xref href=&quot;System.Xaml.XamlInternalException&quot;> </xref> klass med ett meddelande som beskriver fel som tillhandahålls av systemet."
  syntax:
    content: public XamlInternalException ();
    parameters: []
  overload: System.Xaml.XamlInternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Xaml.XamlInternalException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Xaml.XamlInternalException
  langs:
  - csharp
  name: XamlInternalException(String)
  nameWithType: XamlInternalException.XamlInternalException(String)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(String)
  type: Constructor
  assemblies:
  - System.Xaml
  namespace: System.Xaml
  summary: "Initierar en ny instans av den <xref href=&quot;System.Xaml.XamlInternalException&quot;> </xref> klass med ett angivet meddelande som beskriver felet."
  syntax:
    content: public XamlInternalException (string message);
    parameters:
    - id: message
      type: System.String
      description: "Det meddelande som beskriver undantaget."
  overload: System.Xaml.XamlInternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Xaml.XamlInternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Xaml.XamlInternalException
  langs:
  - csharp
  name: XamlInternalException(SerializationInfo,StreamingContext)
  nameWithType: XamlInternalException.XamlInternalException(SerializationInfo,StreamingContext)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - System.Xaml
  namespace: System.Xaml
  summary: "Initierar en ny instans av den <xref href=&quot;System.Xaml.XamlInternalException&quot;> </xref> klass med serialiserade data."
  syntax:
    content: protected XamlInternalException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "Det objekt som innehåller objektdata som serialiserade."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "Detaljerad information om käll- och målservrar."
  overload: System.Xaml.XamlInternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Xaml.XamlInternalException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Xaml.XamlInternalException
  langs:
  - csharp
  name: XamlInternalException(String,Exception)
  nameWithType: XamlInternalException.XamlInternalException(String,Exception)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(String,Exception)
  type: Constructor
  assemblies:
  - System.Xaml
  namespace: System.Xaml
  summary: "Initierar en ny instans av den <xref href=&quot;System.Xaml.XamlInternalException&quot;> </xref> klass med ett angivet meddelande och en referens till det interna undantaget som är orsaken till det här undantaget."
  syntax:
    content: public XamlInternalException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "Det meddelande som beskriver undantaget."
    - id: innerException
      type: System.Exception
      description: "Undantag som är orsaken till det aktuella undantaget. Om den `innerException` parametern är inte <xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;> </xref>, det aktuella undantaget utlöses i en <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;> </xref> block som hanterar det interna undantaget."
  overload: System.Xaml.XamlInternalException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.Xaml.XamlException
  isExternal: false
  name: System.Xaml.XamlException
- uid: System.Xaml.XamlInternalException.#ctor
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException()
  nameWithType: XamlInternalException.XamlInternalException()
  fullName: System.Xaml.XamlInternalException.XamlInternalException()
- uid: System.Xaml.XamlInternalException.#ctor(System.String)
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException(String)
  nameWithType: XamlInternalException.XamlInternalException(String)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Xaml.XamlInternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException(SerializationInfo,StreamingContext)
  nameWithType: XamlInternalException.XamlInternalException(SerializationInfo,StreamingContext)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Xaml.XamlInternalException.#ctor(System.String,System.Exception)
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException(String,Exception)
  nameWithType: XamlInternalException.XamlInternalException(String,Exception)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Xaml.XamlInternalException.#ctor*
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException
  nameWithType: XamlInternalException.XamlInternalException
